lib/json_schema/versioned_schema_reader.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module JsonSchema
+  class VersionedSchemaReader < ::JSON::Schema::Reader
+    def read(schema_uri)
+      super(::JSON::Util::URI.normalized_uri(uri_to_file(schema_uri.path)))
+    end
+
+  private
+
+    attr_reader :version
+
+    def initialize(version: 1.0, options: {})
+      @version = version
+      super(options)
+    end
```

Comment:
Is this meant to be private?

How is this class used exactly?

Reply:
https://www.rubydoc.info/gems/json-schema/2.6.2/JSON/Schema/Reader

It's passed as an option into JSON::Schema.fully_validate to allow it to read the included files when they are declared as references or schemas.